docs/src/references/mqtt-topics.md

```diff
@@ -4,15 +4,411 @@ tags: [Reference, MQTT]
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/main///m/environment' '{
+  "temperature": 23.4
+}'
+```
+
+If the there is no measurement type, then the type can be left empty, but it must have the trailing slash `/` (so that the number of topic segments is the same).
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/main///m/' '{
+  "temperature": 23.4
+}'
+```
+
+#### Publish to a child device
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/child01///m/environment' '{
+  "temperature": 23.4
+}'
+```
+
+#### Publish to a service on the main device
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/main/service/nodered/m/environment' '{
+  "temperature": 23.4
+}'
+```
+
+Any MQTT client can subscribe to all measurements for all entities (devices and services) using the following MQTT topic:
+
+```sh te2mqtt
+tedge mqtt sub 'te/+/+/+/+/m/+'
+```
+
+If you want to be more specific and only subscribe to the main device, then you can used fixed topic names rather than wildcards:
+
+```sh te2mqtt
+tedge mqtt sub 'te/device/main/+/+/m/+'
+```
+
+Or to subscribe to a specific type of measurement published to an services on the main device, then use:
+
+```sh te2mqtt
+tedge mqtt sub 'te/device/main/service/+/m/memory'
+```
+
+#### Publish to a service on a child device
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/child01/service/nodered/m/environment' '{
+  "temperature": 23.4
+}'
+```
+
+### Telemetry type metadata
+
+The data types also may have additional metadata associated with it,
+which can be added/updated by publishing to `/meta` subtopics of those data types.
+For example, the units associated with measurements in the `battery_reading` measurement type
+can be updated by publishing the following message:
+
+```sh te2mqtt
+tedge mqtt pub -r te/device/main///m/battery_reading/meta '{
+  "units": {
+    "temperature": "C",
+    "voltage": "V",
+    "current": "A"
+  }
+}'
+```
+
+The metadata fields supported by each data type will be defined in detail later.
+
+## Commands
+
+The topic scheme for commands is as follows:
+
+```text
+<entity_id_prefix>/cmd/<cmd-type>/<cmd-id>
+```
+
+|Type|Example|
+|------|-------|
+|Software List|`<entity_id_prefix>/cmd/software_list/<cmd-id>`|
+|Software Update|`<entity_id_prefix>/cmd/software_update/<cmd-id>`|
+|Configuration Snapshot|`<entity_id_prefix>/cmd/config_snapshot/<cmd-id>`|
+|Configuration Update|`<entity_id_prefix>/cmd/config_update/<cmd-id>`|
+|Firmware Update|`<entity_id_prefix>/cmd/firmware_update/<cmd-id>`|
+|Restart|`<entity_id_prefix>/cmd/restart/<cmd-id>`|
+
+The command would be interpreted differently based on the target entity.
+For example, the `restart`` could mean either a device restart or a service restart based on the target entity.
+
+### Examples: With default device/service topic semantics
+
+#### Command to main device
+
+Command to fetch the software list from the main device:
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/main/cmd/software_list/123' '{
```

Comment:
What is the difference without/with two slashes?
Here: `te/device/main/cmd/software_list/123`
Config type declaration: `te/device/main///cmd/config_snapshot`

Reply:
This is a mistake, I will correct it, it should have been:

```sh
te/device/main///cmd/software_list/123
```